File: RandomUser/app/src/main/java/com/ztask/sarvan/randomuser/ui/userlist/UserListAdapter.java
package com.ztask.sarvan.randomuser.ui.userlist;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.ztask.sarvan.randomuser.R;
import com.ztask.sarvan.randomuser.data.localdb.entity.RandomUserEntities;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.ThreadLocalRandom;


public class UserListAdapter extends RecyclerView.Adapter<UserListAdapter.ViewHolder> {
    private OnItemClickListener listener;
    private List<RandomUserEntities> results;
    private UserListFragment fragment;
    private List<Integer> dynamicSizes = new ArrayList();
    public UserListAdapter(UserListFragment fragment, List<RandomUserEntities> results) {
        this.fragment = fragment;
        this.listener = fragment;
        this.results = results;
        dynamicSizes.add(320);
        dynamicSizes.add(440);
        dynamicSizes.add(360);
        dynamicSizes.add(400);
        dynamicSizes.add(430);
        dynamicSizes.add(350);
        dynamicSizes.add(370);
    }

    public void setFilter(List<RandomUserEntities> temp) {
        results = temp;
        notifyDataSetChanged();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView tv, tvUserAge;
        ImageView iv;

        ViewHolder(View itemView) {
            super(itemView);
            tv = itemView.findViewById(R.id.itemusenametv);
            tvUserAge = itemView.findViewById(R.id.tvUserAge);
            iv = itemView.findViewById(R.id.itemuserlogoiv);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    listener.OnItemClick(results.get(getAdapterPosition()));
                }
            });
        }
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_user, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        String name = results.get(position).getTitle() +". "+
                results.get(position).getFirst() +" "+
                results.get(position).getLast();
        String imgurl = results.get(position).getUserimg();

        holder.tvUserAge.setText("Age: "+ results.get(position).getAge());

        int random = ThreadLocalRandom.current().nextInt(0, 6);
        int height = dynamicSizes.get(random);
        holder.iv.getLayoutParams().height = height;
        holder.tv.setText(name);
        Glide.with(fragment)
                .load(imgurl)
                .placeholder(R.drawable.ic_randomuser_img) //placeholder
                .error(R.drawable.ic_randomuser_img) //error
                .into(holder.iv);
    }

    @Override
    public int getItemCount() {
        return results.size();
    }
    interface OnItemClickListener{
        void OnItemClick(RandomUserEntities item);
    }
}

File: RandomUser/app/src/main/java/com/ztask/me/randomuser/ui/userlist/UserListAdapter.kt
package com.ztask.me.randomuser.ui.userlist

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.CircleCrop
import com.ztask.me.randomuser.data.localdb.ResultResponseElement
import com.ztask.me.randomuser.data.localdb.ResultsItem
import com.ztask.me.randomuser.databinding.ItemCardBinding

class UserListAdapter (private val listener: UserItemListener) : RecyclerView.Adapter<UserViewHolder>() {

    interface UserItemListener {
        fun onClickedUser(item: ResultsItem)
    }

    private var items = ArrayList<ResultsItem>()

    fun setItems(items: ArrayList<ResultsItem>) {
        this.items.clear()
        this.items.addAll(items)
        notifyDataSetChanged()
    }

    fun filterList(resultList: ArrayList<ResultsItem>) {
        this.items = resultList;
        notifyDataSetChanged();
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
        val binding: ItemCardBinding = ItemCardBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return UserViewHolder(binding, listener)
    }

    override fun getItemCount(): Int = items.size

    override fun onBindViewHolder(holder: UserViewHolder, position: Int) = holder.bind(items[position])
}

class UserViewHolder(private val itemBinding: ItemCardBinding, private val listener: UserListAdapter.UserItemListener) : RecyclerView.ViewHolder(itemBinding.root),
        View.OnClickListener {

    private lateinit var user: ResultsItem

    init {
        itemBinding.root.setOnClickListener(this)
    }

    @SuppressLint("SetTextI18n")
    fun bind(item: ResultsItem) {
        this.user = item
        itemBinding.ivCountryName.text = "${item.name?.title}. ${item.name?.first} ${item.name?.last}"
        Glide.with(itemBinding.root)
                .load(item.picture?.large)
                .transform(CircleCrop())
                .into(itemBinding.ivFlag)
    }

    override fun onClick(v: View?) {
        listener.onClickedUser(user)
    }
}